os/kernel/binary_manager/binary_manager_bootparam.c

```diff
@@ -144,6 +144,7 @@ int binary_manager_scan_bootparam(binmgr_bpinfo_t *bp_info)
 			printf("[Bootparam Scanning] Failed to read bootparam, errno %d\n", errno);
 			continue;
 		} else if (!is_valid_bootparam(bootparam)) {
+			printf("[Bootparam Scanning] Invalid BP %d\n", bp_idx);
```

Comment:
I think it is better to use `printf` from `is_valid_bootparam`, because there are reasons why bootparam is invalid.